igd/__main__.py

```diff
@@ -6,7 +6,9 @@
 import socket
 
 from . import core, proto
-
+selector = selectors.DefaultSelector()
```

Comment:
Seems like this was fixed in `curio` itself: https://github.com/dabeaz/curio/pull/159
Could you try just upgrading `curio`? 
I don't ghave IGD enabled router now, neither Windows machine.